baseplate/clients/requests.py

```diff
@@ -216,7 +223,9 @@ def send(self, request: PreparedRequest, **kwargs: Any) -> Response:
             session.mount("https://", self.adapter)
             response = session.send(request, **kwargs)
 
-            span.set_tag("http.status_code", response.status_code)
+            http_status_code = response.status_code
+            span.set_tag("http.status_code", http_status_code)
+            span.set_tag("http.success", str(200 <= http_status_code < 400).lower())
```

Comment:
For consistency, let's mirror the HTTP server metrics and infer success from the status code rather than setting it as a tag here.

https://github.com/reddit/baseplate.py/blob/a94f21f7d4d229757572d2f2fc714b543f9a386e/baseplate/lib/prometheus_metrics.py#L203-L207

Reply:
I kinda disagree. The success is not a prometheus thing. Either the request is successful or it's not. No need to do this multiple times per request (for each metric) when the client should be responsible to tell us whether the query is successful or not.